eng/pipelines/templates/steps/build.yml

```diff
@@ -0,0 +1,200 @@
+      parameters:
+        ServiceDirectory: ${{ parameters.ServiceDirectory }}
+
+  - template: /eng/common/pipelines/templates/steps/daily-dev-build-variable.yml
+
+  - template: /eng/pipelines/templates/steps/set-artifact-packages.yml
+    parameters:
+      PackageInfo: '$(Build.ArtifactStagingDirectory)/PackageInfo'
+      Artifacts: ${{ parameters.Artifacts }}
+
+  - pwsh: |
+      # Reset SkipDevBuildNumber as empty
+      $skipDevBuildNumber = ""
+      echo "##vso[task.setvariable variable=SkipDevBuildNumber]$skipDevBuildNumber"
+      Write-Host "SkipDevBuildNumber: [$(SkipDevBuildNumber)]"
+      $skipDevBuildNumber = "false"
+      # For .NET builds the only case we want to not have dev build numbers on our packages is when it is manually queued and
+      # the SetDevVersion property isn't set to true. All other cases should contain dev version numbers.
+      if ('$(Build.Reason)' -eq 'Manual' -and '$(SetDevVersion)' -ne 'true') {
+        $skipDevBuildNumber = "true"
+      }
+      $versioningProperties = "/p:OfficialBuildId=$(OfficialBuildId) /p:SkipDevBuildNumber=$skipDevBuildNumber";
+      echo "##vso[task.setvariable variable=VersioningProperties]$versioningProperties"
+    displayName: "Setup .NET specific versioning properties"
+
+  - task: UsePythonVersion@0
+    displayName: 'Use Python 3.11'
+    inputs:
+      versionSpec: '3.11'
+
+  - template: /eng/pipelines/templates/steps/dotnet-diagnostics.yml
+    parameters:
+      LogFilePath: $(Build.ArtifactStagingDirectory)/pack.binlog
+
+  # only use the project list override file if the service directory is not auto
+  - ${{ if eq(parameters.ServiceDirectory, 'auto') }}:
+    - pwsh: |
+        if (!"$(ProjectNames)") {
+          Write-Host "No packages to build."
+          exit 0
+        }
+        dotnet pack eng/service.proj -warnaserror `
+          /p:ValidateRunApiCompat=true `
+          /p:SDKType=${{ parameters.SDKType }} `
+          /p:IncludeTests=false `
+          /p:PublicSign=false $(VersioningProperties) `
+          /p:Configuration=$(BuildConfiguration) `
+          /p:CommitSHA=$(Build.SourceVersion) `
+          /p:ArtifactsPackagesDir=$(Build.ArtifactStagingDirectory) `
+          /p:ServiceDirectory=* `
+          /p:ProjectListOverrideFile=$(ProjectListOverrideFile) `
+          /p:IncludePerf=false `
+          /p:IncludeStress=false `
+          /p:IncludeIntegrationTests=false `
+          /p:EnableOverrideExclusions=true `
+          $(DiagnosticArguments)
+      displayName: "Build and Package for PR"
+  - ${{ else }}:
+    - pwsh: |
+        $variableSet = $env:ProjectListOverrideFile -ne "`$`(ProjectListOverrideFile`)"
+        if ($variableSet -and (Test-Path $env:ProjectListOverrideFile)) {
+          Write-Host "Clearing $env:ProjectListOverrideFile"
+          rm $env:ProjectListOverrideFile
+        }
+        Write-Host '##vso[task.setvariable variable=ProjectListOverrideFile;]'
```

Comment:
I'm not sure I'm understanding why you are clearing this here as it seems to be used below in the next step. Couldn't we accomplish the same thing by explicitly passing empty or not passing it at all in the next step?